gitlabbot/gitlabbot/db.go

```diff
@@ -0,0 +1,166 @@
+package gitlabbot
+
+import (
+	"database/sql"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+
+	"github.com/keybase/managed-bots/base"
+	"golang.org/x/oauth2"
+)
+
+type DB struct {
+	*base.DB
+}
+
+func NewDB(db *sql.DB) *DB {
+	return &DB{
+		DB: base.NewDB(db),
+	}
+}
+
+// webhook subscription methods
+
+func (d *DB) CreateSubscription(convID chat1.ConvIDStr, repo string, branch string, hookID int64) error {
```

Comment:
Would be nice if this code didn't have to be repeated verbatim from githubbot, changes will have to be reflected i both bots. Could the storage layer be abstracted so both bots can share? cc @hyperobject

Reply:
Probably wouldn't make sense to share this code since githubbot is moving to the gh apps api which has different subscription ids

Reply:
Yeah, I think it also gives more flexibility if the two bots diverge